src/decompose_mesh/module_database.f90

```diff
@@ -0,0 +1,457 @@
+                kE = ie_bnd_stored(inum_neigh_part)
+                my_interfaces_ext_mesh(1,kE ,inum_neigh_part) = glob2loc_elmnt(iE)
+                my_interfaces_ext_mesh(2,kE ,inum_neigh_part) = k 
+                my_interfaces_ext_mesh(3:6,kE ,inum_neigh_part) = liste_comm_nodes(1:NGNOD2D)
+             end if
+          end do
+       end if
+    end do
+  
+    write(IIN_database) nb_stored_slice, max_element_b_partition
+    do islice = 1, nb_stored_slice
+       !write(27,*) 'ii ' , myrank, islice_neigh(islice)-1, num_element_in_boundary_partition(islice_neigh(islice))
+       write(IIN_database) islice_neigh(islice)-1, num_element_in_boundary_partition(islice_neigh(islice))
+       do iE = 1, num_element_in_boundary_partition(islice_neigh(islice))
+          write(IIN_database)  my_interfaces_ext_mesh(1:6, iE, islice)
+       end do
+    end do
+
+    ! write MOHO 
+    ! optional moho
+    loc_nspec2D_moho = 0
+    do i=1,nspec2D_moho
+       if (ipart(ibelm_moho(i)) == myrank+1) then
+          loc_nspec2D_moho = loc_nspec2D_moho + 1
+       endif
+    enddo
+    ! checks if anything to do
+    if (loc_nspec2D_moho > 0) then
+
+       ! format: #surface_id, #number of elements
+       write(IIN_database) 7, loc_nspec2D_moho
+       
+       ! outputs element index and element node indices
+       ! note: assumes that element indices in ibelm_* arrays are in the range from 1 to nspec
+       !          (this is assigned by CUBIT, if this changes the following indexing must be changed as well)
+       !          while glob2loc_elmnts(.) is shifted from 0 to nspec-1  thus
+       !          we need to have the arg of glob2loc_elmnts start at 0 ==> glob2loc_nodes(ibelm_** -1)
+       
+       ! optional moho
+       do i=1,nspec2D_moho
+          if (ipart(ibelm_moho(i)) == myrank+1) then
+             do inode = 1,NGNOD2D
+                node_loc(inode) = glob2loc_nodes(nodes_ibelm_moho(inode,i))
+             enddo
+             write(IIN_database) glob2loc_elmnt(ibelm_moho(i)), node_loc(1:NGNOD2D)
+          endif
+       enddo
+    end if
+    close(IIN_database)
+    
+  end subroutine write_database
+
+!----------------------------------
+!
+!  count elements in boudary
+!
+!----------------------------------
+
+  subroutine count_my_boundary(myrank, ipart, ibelm, nspec2D, nE, iflag)
+
+    implicit none
+    integer,                                   intent(in) :: myrank, iflag
+    integer,                                   intent(in) :: nspec2D, nE
+    integer,   dimension(nE),                  intent(in) :: ipart
+    integer,   dimension(nspec2D),             intent(in) :: ibelm
+
+    integer ::  i, nspec2D_loc, iE
+    
+    nspec2D_loc = 0
+    do i=1, nspec2D
+       iE=ibelm(i)
+       if (ipart(iE) == myrank +1) nspec2D_loc = nspec2D_loc + 1
+    end do
+    write(IIN_database) iflag, nspec2D_loc
+
+  end subroutine count_my_boundary
+
+!----------------------------------
+!
+!  write elements in boudary
```

Comment:
Fixed.

On 02/10/2016 07:29 PM, Elliott Sales de Andrade wrote:
> In src/decompose_mesh/module_database.f90
> <https://github.com/geodynamics/specfem3d/pull/694#discussion_r52501445>:
>
>> +    integer,   dimension(nspec2D),             intent(in) :: ibelm
>> +
>> +    integer ::  i, nspec2D_loc, iE
>> +
>> +    nspec2D_loc = 0
>> +    do i=1, nspec2D
>> +       iE=ibelm(i)
>> +       if (ipart(iE) == myrank +1) nspec2D_loc = nspec2D_loc + 1
>> +    end do
>> +    write(IIN_database) iflag, nspec2D_loc
>> +
>> +  end subroutine count_my_boundary
>> +
>> +!----------------------------------
>> +!
>> +!  write elements in boudary
>
> boudary -> boundary
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem3d/pull/694/files#r52501445>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr